[New Content] - Overture Guides (#385)
overture-stack/website: markdown/documentation/guides/administration/portalcustomization.md

---
title: Customizing your Data Portal
---

# Getting Started

## Introduction

> Arrangers simplifies graphQL queries over elasticsearch indices with it's front-end library of search components. The primary configurable components for this guide are the left hand search facet and the central data table seen below.

Reviewer: ```suggestion
Arranger simplifies GraphQL queries over Elasticsearch indices with it's front-end library of reusable search components. The primary configurable components for this guide are the left hand search facets and the central data table seen below.
```

![Portal](./assets/portal.png 'Portal')

<Note title="Arranger Configration Files">All configurations for these components are made through four configuration files: `base.json`, `extended.json`, `table.json` and `facets.json`. We will cover each in the following sections.</Note>

## Viewing Elasticsearch Documents

Indices in Elasticsearch are a collection of related documents with similar characteristics.

> **ElasticVue** offers a convenient and user-friendly interface for managing and exploring your Elasticsearch data. With ElasticVue, you can:

Reviewer: Fairly confident they use all lowercase for their product, and we can link to their page for proper credit and for people to read up on it themselves.

```suggestion
**[Elasticvue](https://elasticvue.com/)** offers a convenient and user-friendly interface for managing and exploring your Elasticsearch data. With elasticvue, you can:
```

- Easily visualize and search through indexed documents.
- Quickly access and interact with JSON documents.
- Simplify the management and troubleshooting of Elasticsearch indices.

> **To install ElasticVue, follow these steps:**

Reviewer: ```suggestion
**To install elasticvue, follow these steps:**
```

> 1. Visit the Chrome Web Store (or Firefox Add-ons) and search for the **ElasticVue** browser extension.

Reviewer: ```suggestion
1. Search for **elasticvue**  in your browser's extension catalogue. E.g. Chrome Web Store, Firefox Add-ons, Microsoft Webstore.
```

2. Click on **"Add to Chrome"** (or **"Add to Firefox"**) to install the extension.

> 3. Open ElasticVue and enter your Elasticsearch URL. For the Overture Quickstart, this will be `localhost:9200`.

Reviewer: ```suggestion
3. Open elasticvue and enter your Elasticsearch URL. For the Overture Quickstart, this will be `http://localhost:9200`.
```

4. Select basic authentication and enter the default username `elastic` and password `myelasticpassword`.

### Using ElasticVue

> From the ElasticVue dashboard's top navigation, select **search**.
> 
> ![ElasticVue](./assets/elasticvue.png 'ElasticVue')

Reviewer: ```suggestion
From the elasticvue dashboard's top navigation, select **search**.

![Elasticvue](./assets/elasticvue.png 'Elasticvue')
```

This page displays all indexed Elasticsearch documents created by Maestro from published Song analyses and used by Arranger. Clicking any of the `_index` rows will give you a direct view of the JSON documents that populate the index.

Being able to easily view the JSON documents within your elastic search instance will be beneficial when configuring your Arranger configs.

# Arranger Configurations

## Base Configuration

The [base.json file](https://github.com/overture-stack/composer/blob/develop/configurationFiles/arrangerConfigs/base.json) contains only two fields, `documenType` and `index`: 

```json
{
	"documentType": "file",
	"index": "overture-quickstart-index"
}
```

-  The `index` field specifies the name of the Elasticsearch index, in this example the `overture-quickstart-index`

> -  `documentType` informs arranger of the mapping type being used by Maestro, `analysis` or `file` centric

Reviewer: ```suggestion
-  `documentType` informs Arranger of the mapping type being used by Maestro, `analysis` or `file` centric
```

## Extended Configuration

The extended.json configuration file defines all the fields and display names you wish to populate your front-end portal with. Below, we have provided a simplified list taken from our [QuickStart extended.json](https://github.com/overture-stack/composer/blob/develop/configurationFiles/arrangerConfigs/extended.json) configuration:

```JSON
{
	"extended": [
	  {
		"displayName": "Object Id",
		"fieldName": "object_id"
	  },
    {
		"displayName": "Analysis Id",
		"fieldName": "analysis.analysis_id"
	  },
    {
		"displayName": "Treatment Duration (Days)",
		"fieldName": "analysis.donor.primaryDiagnosis.treatment.treatmentDuration"
	  }
  ]
}

```

The`displayName` field outlines how you want your fields displayed on the front-end UI when used within the search facets and or table.

The `fieldName` values are written as represented within your Elasticsearch documents:

- Object ID can be found at the root of the [Elasticsearch Documents](https://github.com/overture-stack/composer/blob/develop/guideMaterials/dataAdministration/ES-fileCentric-document.json#L217) and therefore is simply the fieldName shown here

> - The Analysis Id is a nested element [found inside the Analysis field](https://github.com/overture-stack/composer/blob/develop/guideMaterials/dataAdministration/ES-fileCentric-document.json#L61C1-L80C61), we denote nesting by adding a period `.` making the appropriate fieldName `analysis.analysis_id`

Reviewer: Another consistent spelling check: Id vs ID.

```suggestion
- The Analysis ID is a nested element [found inside the Analysis field](https://github.com/overture-stack/composer/blob/develop/guideMaterials/dataAdministration/ES-fileCentric-document.json#L61C1-L80C61), we denote nesting by adding a period `.` making the appropriate fieldName `analysis.analysis_id`
```
